app/views/flexible_criteria/index.html.erb

```diff
@@ -9,16 +9,17 @@
     <%= t('criteria.marking_scheme', identifier: @assignment.short_identifier) %>
   </h1>
   <div class='heading_buttons'>
-    <%= link_to t('add_criterion'), 
-          'javascript:void(0);',
-          onclick: "if(document.getElementById('new_criterion') != null) {
-          document.getElementById('new_criterion_prompt').focus();
-        } else {
-          jQuery.ajax({
-            url: '" +new_assignment_flexible_criterion_path(@assignment.id) + "',
-            type: 'GET'
-          });
-        }"%>
+    <%= link_to t('criteria.add_criterion'),
+      'javascript:void(0);',
+      onclick: "if(document.getElementById('new_criterion') != null) {
```

Comment:
Please make this a separate function in one of the erb files while you're changing this.

Make the criterion type a parameter to this function, so when you call it here with the onclick, you pass in either "FlexibleCriterion" or "RubricCriterion"